### PR TITLE
feat: add observe_monitor_v2_action resource

### PR DESCRIPTION
**Pull request by obs-gh-owengoebel**

### CODE REVIEW

File: observe/resource_monitor_v2_action.go [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+
+func resourceMonitorV2Action() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMonitorV2Create,
+		ReadContext:   resourceMonitorV2Read,
+		UpdateContext: resourceMonitorV2Update,
+		DeleteContext: resourceMonitorV2Delete,
+		Schema: map[string]*schema.Schema{
+			// needed as input to CreateMonitorV2Action
+			"workspace_id": { // ObjectId!
```
Comment (vikramraman) [inline observe/resource_monitor_v2_action.go:25]: just `workspace`

File: observe/resource_monitor_v2_action.go [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+				Type:             schema.TypeString,
+				ForceNew:         true,
+				Required:         true,
+				ValidateDiagFunc: validateOID(oid.TypeWorkspace),
+			},
+			// fields of MonitorV2ActionInput
+			"inline": { // Boolean
```
Comment (vikramraman) [inline observe/resource_monitor_v2_action.go:32]: I believe we only want to support shared actions. So perhaps we should not expose this, and default to false internally.

File: observe/resource_monitor_v2_action.go [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+			"icon_url": { // String
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": { // String
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			// end of monitorV2ActionInput
+			"id": { // ObjectId!
```
Comment (vikramraman) [inline observe/resource_monitor_v2_action.go:66]: `oid` instead of `id`

File: observe/resource_monitor_v2_action.go [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+		},
+	}
+}
+
+func monitorV2EmailActionInput() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"subject": { // String
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (vikramraman) [inline observe/resource_monitor_v2_action.go:79]: Should the `subject` and `body` be required for an email action? Feels odd that all of these are optional for an email input.

File: observe/resource_monitor_v2_action.go [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+		},
+	}
+}
+
+func monitorV2WebhookActionInput() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"headers": { // [MonitorV2WebhookHeaderInput!]
+				Type:     schema.TypeList,
+				Optional: true,
```
Comment (vikramraman) [inline observe/resource_monitor_v2_action.go:100]: Similar question here, should these be required for a webhook input?

File: observe/resource_monitor_v2_action.go [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+			},
+			"type": { // MonitorV2ActionType!
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validateEnums(gql.AllMonitorV2ActionTypes),
+				Required:         true,
+			},
+			"email": { // MonitorV2EmailDestinationInput
+				Type:     schema.TypeList,
+				MaxItems: 1,
```
Comment (vikramraman) [inline observe/resource_monitor_v2_action.go:43]: Should add `ExactlyOneOf: []string{"email", "webhook"}` here and for webhook?

File: observe/resource_monitor_v2_action.go [hunk 7]
```diff
@@ -0,0 +1,390 @@
...
+func resourceMonitorV2ActionDelete(ctx context.Context, data *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
+	client := meta.(*observe.Client)
+	if err := client.DeleteMonitorV2Action(ctx, data.Id()); err != nil {
+		return diag.Errorf("failed to delete monitor action: %s", err.Error())
+	}
+	return diags
+}
+
+func resourceMonitorV2ActionRead(ctx context.Context, data *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
+	// TODO
```
Comment (vikramraman) [inline observe/resource_monitor_v2_action.go:182]: Remove the TODO?

**Review by vikramraman — CHANGES_REQUESTED** [review]

File: observe/resource_monitor_v2_action.go (hunk 4 above)
Comment (obs-gh-owengoebel) [reply observe/resource_monitor_v2_action.go:79]: They are optional in the schema. Also in general emails don't need to have a subject and a body. I can send an email to myself without specifying either

File: observe/resource_monitor_v2_action.go (hunk 5 above)
Comment (obs-gh-owengoebel) [reply observe/resource_monitor_v2_action.go:100]: the schema declares these as [MonitorV2WebhookHeader!], which means the user is not required to provide any elements

Comment (obs-gh-owengoebel) [reply observe/resource_monitor_v2_action.go:97]: Resolved. The API fills with defaults if not provided.

File: observe/resource_monitor_v2_action.go (hunk 4 above)
Comment (obs-gh-owengoebel) [reply observe/resource_monitor_v2_action.go:76]: Resolved. The API fills with defaults if not provided.

**Review by vikramraman — APPROVED** [review]

**Pull request merged**